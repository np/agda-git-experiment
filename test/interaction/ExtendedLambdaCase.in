top_command (cmd_load currentFile [])

-- Normalise and display a lambda lifted  extended lambda
top_command (cmd_compute_toplevel False "f")

-- Refine for extended lambdas (issue 713)
goal_command 4 cmd_refine "λ { a {x} b → a }"

-- Case splitting for function with implicit argument
top_command (showImplicitArgs True)
goal_command 3 cmd_make_case "z"
goal_command 3 cmd_make_case "w"
top_command (showImplicitArgs False)

-- Case splitting with lambda lifted definition, hidden arguments
goal_command 2 cmd_make_case "x'"
top_command (cmd_load currentFile [])
top_command (showImplicitArgs True)
goal_command 2 cmd_make_case "x'"


-- Case splitting with lambda lifted definition
goal_command 1 cmd_make_case "x"

-- Case splitting for nested extended lambdas
goal_command 0 cmd_make_case "y"

