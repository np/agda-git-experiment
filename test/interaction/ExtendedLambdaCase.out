Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking ExtendedLambdaCase (ExtendedLambdaCase.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished ExtendedLambdaCase.\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Bool\n?1 : Bool\n?2 : Bool\n?3 : Bool\n?4 : {A : Set} → A → {A} → A → A\n" nil)
((last . 1) . (agda2-goals-action '(0 1 2 3 4)))
Agda2> Agda2> Agda2> (agda2-status-action "")
(agda2-info-action "*Normal Form*" "(y : Bool) →\nparameterised.Bar (λ _ → Void) ((λ {true → true; false → false}) y)" nil)
((last . 1) . (agda2-goals-action '(0 1 2 3 4)))
Agda2> Agda2> Agda2> (agda2-give-action 4 "λ {a {x} b → a}")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Bool\n?1 : Bool\n?2 : Bool\n?3 : Bool\n" nil)
((last . 1) . (agda2-goals-action '(0 1 2 3)))
Agda2> Agda2> Agda2> (agda2-status-action "ShowImplicit")
((last . 1) . (agda2-goals-action '(0 1 2 3)))
Agda2> ((last . 2) . (agda2-make-case-action-extendlam '("true {w} → ?" "false {w} → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3)))
Agda2> ((last . 2) . (agda2-make-case-action-extendlam '("z {true} → ?" "z {false} → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3)))
Agda2> (agda2-status-action "")
((last . 1) . (agda2-goals-action '(0 1 2 3)))
Agda2> Agda2> Agda2> ((last . 2) . (agda2-make-case-action-extendlam '("true → ?" "false → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3)))
Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking ExtendedLambdaCase (ExtendedLambdaCase.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished ExtendedLambdaCase.\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Bool\n?1 : Bool\n?2 : Bool\n?3 : Bool\n?4 : {A : Set} → A → {A} → A → A\n" nil)
((last . 1) . (agda2-goals-action '(0 1 2 3 4)))
Agda2> (agda2-status-action "ShowImplicit")
((last . 1) . (agda2-goals-action '(0 1 2 3 4)))
Agda2> ((last . 2) . (agda2-make-case-action-extendlam '("true → ?" "false → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3 4)))
Agda2> Agda2> Agda2> Agda2> ((last . 2) . (agda2-make-case-action-extendlam '("true → ?" "false → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3 4)))
Agda2> Agda2> Agda2> ((last . 2) . (agda2-make-case-action-extendlam '("true z → ?" "false z → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3 4)))
Agda2> Agda2> 